src/v3/src/components/InputPassword/InputPassword.tsx

```diff
@@ -13,30 +13,43 @@
 import { Box, FormHelperText } from '@mui/material';
 import { PasswordInput } from '@okta/odyssey-react-mui';
 import { h } from 'preact';
-import { ChangeEvent, FieldElement, UISchemaElementComponent } from 'src/types';
 
-import { getMessage } from '../../../../v2/ion/i18nTransformer';
 import { useOnChange, useValue } from '../../hooks';
+import {
+  ChangeEvent,
+  UISchemaElementComponent,
+  UISchemaElementComponentWithValidationProps,
+} from '../../types';
 import { getLabelName } from '../helpers';
+import { withFormValidationState } from '../hocs';
 
-const InputPassword: UISchemaElementComponent<{
-  uischema: FieldElement
-}> = ({ uischema }) => {
+const InputPassword: UISchemaElementComponent<UISchemaElementComponentWithValidationProps & {
+  onChange?: (e: ChangeEvent<HTMLInputElement>) => void
+}> = ({
+  uischema,
+  setTouched,
+  error,
+  setError,
+  onValidateHandler,
+  onChange,
+}) => {
   const value = useValue(uischema);
   const onChangeHandler = useOnChange(uischema);
```

Comment:
Is there any way to change the signature of either `useOnChange` or `onChange` to be the same (i.e. both `ChangeEvent` or both `string`) so that this can be:
```js
const onChangeHandler = typeof onChange === 'function' ? onChange : useOnChange(uischema);
```

Reply:
I really like this approach but unless I do something like the below, it is not allowed, see: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level
Specifically: `Don’t call Hooks inside loops, conditions, or nested functions.`

```
const changeHandler = useOnChange(uischema);
const onChangeHandler = typeof onChange === 'function' ? onChange : changeHandler;
```

Reply:
I assumed the above would be acceptable so I pushed it, if not just let me know and I will revert.

Reply:
Oh right, ok I forgot about `rules of hooks`. The change looks good to me.